protos/flyteidl/core/types.proto

```diff
@@ -56,6 +56,14 @@ message BlobType {
     BlobDimensionality dimensionality = 2;
 }
 
+// Enables declaring enum types, with predefined string values
+message EnumType {
+    // Predefined set of enum values
+    repeated string values = 1;
+    // default value for the enum type
+    string default_value = 2;
```

Comment:
Should the default value just be the first value?

Reply:
I wanted to have a way of not having a default. Thus in the drop down no value could be selected

Reply:
I actually like the protobuf decision and golang's of having the first be the default and forcing people, if they want, to name the first one UNDEFINED... 

I think it also simplifies UI implementation and other sdk languages...

Reply:
I actually think protobuf's decision was based on the wire-format and that `0` value is the default int value.
I am thinking of cases, where you do not want a value to be selected at all. I understand we can define a `UNDEFINED` value, but this for users - who never use protobuf (think data scientists, data engineers) and users who use python - which supports None value, would be harder to understand.

But I am ok with whatever, as I think we can create the representation in higher levels. Just that ever ux layer wil need to represent it in the same way

Reply:
Let me just remove the default value, we can always add that in